### PR TITLE
feat: add global image settings for all images in the EG chart

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR exposes a top-level global image configuration in the helm value file for images used in the EG helm chart.

```
global:
  images:
    envoyGateway:
      image: "private-hub/envoyproxy/gateway:abcdef12"
      pullPolicy: Always
      imagePullSecrets:
      - name: "secret1"
      - name: "secret2"
    ratelimit:
      image: "private-hub/envoyproxy/ratelimit:test"

```

Future work to support air-gapped envirionment:
* Support `pullPolicy` and `imagePullSecrets` for ratelimit
* Support  custom image setting for statsd-exporter 
* Support custom image setting for envoy

Those will be addressed in follow-up PRs.

part of: https://github.com/envoyproxy/gateway/issues/3049

### CODE REVIEW

File: charts/gateway-helm/values.tmpl.yaml [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+      image: ratelimit
+      tag: 'master'
+
 deployment:
   envoyGateway:
     image:
-      repository: ${ImageRepository}
-      tag: '${ImageTag}'
-    imagePullPolicy: ${ImagePullPolicy}
+      repository: ""
```
Comment (zhaohuabing) [inline charts/gateway-helm/values.tmpl.yaml:22]: Image settings are retained here for backward compatibility.

File: charts/gateway-helm/values.tmpl.yaml [hunk 2]
```diff
@@ -1,9 +1,27 @@
+# The global settings for the Envoy Gateway Helm chart.
+# These values will be used if the values are not overridden in the other sections.
+global:
```
Comment (zhaohuabing) [inline charts/gateway-helm/values.tmpl.yaml:3]: Not sure a `global` umbrella makes sense here. 

If may be other `global` options in the value files, however, the EG value file is not that large.

**Comment by arkodg** [comment]
@zhaohuabing what happens what the image is set in top level as well as deployment, which one wins ? I'd vote for most specific one (deployment) wins

**Comment by zhaohuabing** [comment]
> @zhaohuabing what happens what the image is set in top level as well as deployment, which one wins ? I'd vote for most specific one (deployment) wins

Deployment level wins. Ideally we should remove the deployment level config. I keep it for back compatibility.

File: site/content/en/latest/install/api.md [hunk 3]
```diff
@@ -31,12 +31,11 @@ The Helm chart for Envoy Gateway
 | certgen.rbac.labels | object | `{}` |  |
 | config.envoyGateway.gateway.controllerName | string | `"gateway.envoyproxy.io/gatewayclass-controller"` |  |
 | config.envoyGateway.logging.level.default | string | `"info"` |  |
-| config.envoyGateway.provider.kubernetes.shutdownManager.image | string | `"docker.io/envoyproxy/gateway:latest"` |  |
 | config.envoyGateway.provider.type | string | `"Kubernetes"` |  |
 | createNamespace | bool | `false` |  |
-| deployment.envoyGateway.image.repository | string | `"docker.io/envoyproxy/gateway"` |  |
-| deployment.envoyGateway.image.tag | string | `"latest"` |  |
-| deployment.envoyGateway.imagePullPolicy | string | `"IfNotPresent"` |  |
+| deployment.envoyGateway.image.repository | string | `""` |  |
```
Comment (arkodg) [inline site/content/en/latest/install/api.md:36]: should we delete this fields then ?

File: tools/make/helm.mk [hunk 4]
```diff
@@ -28,9 +28,15 @@ helm-install: helm-generate ## Install envoy gateway helm chart from OCI registr
...
 .PHONY: helm-generate
 helm-generate:
-	ImageRepository=${IMAGE} ImageTag=${TAG} ImagePullPolicy=${IMAGE_PULL_POLICY} envsubst < charts/gateway-helm/values.tmpl.yaml > ./charts/gateway-helm/values.yaml
+	GatewayImage=${IMAGE}:${TAG} GatewayImagePullPolicy=${IMAGE_PULL_POLICY} envsubst < charts/gateway-helm/values.tmpl.yaml > ./charts/gateway-helm/values.yaml
 	helm lint charts/gateway-helm
 
-helm-template: ## Template envoy gateway helm chart.
+HELM_VALUES := $(wildcard test/helm/*.in.yaml)
+
+helm-template: ## Template envoy gateway helm chart.z
```
Comment (arkodg) [inline tools/make/helm.mk:36]: imo lets create a new `helm-test` target instead of reusing this one

Comment (zirain) [reply tools/make/helm.mk:36]: it not a test, because it will alwasy pass. it just help code review.

**Review by arkodg — APPROVED** [review]
LGTM thanks, this will def simplify operations !

**Review by zirain — APPROVED** [review]

File: site/content/en/latest/install/api.md (hunk 3 above)
Comment (zhaohuabing) [reply site/content/en/latest/install/api.md:36]: Just keep this for back compatibility.  If removed, the existing helm value file containing these values won't work, which will surprise users.

**Pull request merged**